src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java

```diff
@@ -15,7 +16,7 @@
     @Override
     public User login(String login, String password) throws AuthenticationException {
         Optional<User> userFromDB = userService.findByLogin(login);
-        if (userFromDB.isEmpty() || !userFromDB.get().getPassword().equals(password)) {
+        if (userFromDB.isEmpty() || !HashUtil.isValid(password, userFromDB.get())) {
```

Comment:
let's move the private method to this class

we pass user to the method
HashUtil should not have methods with such a specific business logic